### PR TITLE
feat(widgets/chart): add option to set the position of legend

### PR DESCRIPTION
**Pull request by orhun**
> Upstream: [#628](https://github.com/fdehau/tui-rs/pull/628)

## Description

Add option to position the legend at corner. The option uses `tui::layout::Corner` emum that already made.

If the layout have axis title, the layout draw the legend position above x axis title or under y axis title.

Related Issue #335 

## Testing guidelines

Currently, not add test.

Chart example with legend position

## Checklist

* [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
* [ ] I have added relevant tests.
* [ ] I have documented all new additions.

### CODE REVIEW

File: src/widgets/chart.rs [hunk 1]
```diff
@@ -226,6 +226,8 @@ pub struct Chart<'a> {
     style: Style,
     /// Constraints used to determine whether the legend should be shown or not
     hidden_legend_constraints: (Constraint, Constraint),
+    /// The position of a legend
+    legend_position: Corner,
```
Comment (joshka) [inline src/widgets/chart.rs:230]: Would we ever want to put a legend say centered bottom, or vertically centered right? If so, what do we need as a positioning enum here?

File: src/widgets/chart.rs [hunk 2]
```diff
@@ -342,12 +360,25 @@ impl<'a> Chart<'a> {
...
+                    ),
+                    Corner::BottomRight => (
+                        layout.graph_area.right() - legend_width,
+                        layout.graph_area.bottom()
+                            - legend_height
+                            - if self.x_axis.title.is_some() { 1 } else { 0 },
+                    ),
+                    Corner::BottomLeft => (layout.graph_area.left(), layout.graph_area.bottom()),
+                };
+                layout.legend_area = Some(Rect::new(x, y, legend_width, legend_height));
```
Comment (joshka) [inline src/widgets/chart.rs:381]: Refactoring suggestion: pull this chunk out into a method that just calculates the position of the legend (for making the code easier to read.

**Review by joshka — COMMENTED** [review]
Needs a unit tests for each corner / position.

File: src/widgets/chart.rs (hunk 1 above)
Comment (foresterre) [reply src/widgets/chart.rs:230]: An alternative could be to draw the legend at certain coordinates, within the graph area, and provide default constructor functions for each of the four corners.

Comment (joshka) [reply src/widgets/chart.rs:230]: That's not a bad idea - what would the configuration of the widget for that look like?

**Comment by joshka** [comment]
In order to complete this PR it requires:
- [ ] Unit tests
- [ ] docs
- [ ] decide how to handle positioning for centered legends (is there an approach that is open to future implementation perhaps?)

Marked as Draft and up for grabs

**Comment by lyuha** [comment]
Is there any update, here?
If no one work now, could I continue this PR?

**Comment by mindoodoo** [comment]
@lyuha No I don't think anybody is working on this. I'd recommend that for simplicity's sake you reopen this PR as this one was imported from the original `tui-rs` project. As soon as you do so I'll close this one :)

**Comment by joshka** [comment]
This issue was closed due to it becoming stale. Feel free to reopen and work on this.
